tools/iree-tracy-print.cc

```diff
@@ -0,0 +1,304 @@
+  // XEvent corresponds to each Zone.
+  std::unordered_map<uint16_t, tensorflow::profiler::XLine*> xlines;
+
+  for (const auto& z : worker.GetSourceLocationZones()) {
+    auto& event_metadata = (*xplane->mutable_event_metadata())[z.first];
+    event_metadata.set_id(z.first);
+    event_metadata.set_name(GetZoneName(worker, z.first));
+    event_metadata.set_display_name(event_metadata.name());
+
+    for (const auto& t : z.second.zones) {
+      auto it = xlines.find(t.Thread());
+      if (it == xlines.end()) {
+        it = xlines.insert(std::make_pair(t.Thread(), xplane->add_lines()))
+             .first;
+        auto* xline = it->second;
+        xline->set_id(t.Thread());
+        xline->set_display_id(t.Thread());
+        xline->set_name(worker.GetThreadName(
+            worker.DecompressThread(t.Thread())));
+        xline->set_display_name(xline->name());
+        //xline->set_timestamp_ns();
+        //xline->set_duration_ps();
+      }
+
+      auto* event = it->second->add_events();
+      event->set_metadata_id(z.first);
+      event->set_offset_ps(t.Zone()->Start() * 1000);
+      event->set_duration_ps((t.Zone()->End() - t.Zone()->Start()) * 1000);
+    }
+  }
+
+  for (const auto& z : worker.GetGpuSourceLocationZones()) {
+    auto& event_metadata = (*xplane->mutable_event_metadata())[z.first];
+    event_metadata.set_id(z.first);
+    event_metadata.set_name(GetZoneName(worker, z.first));
+    event_metadata.set_display_name(event_metadata.name());
+
+    for (const auto& t : z.second.zones) {
+      auto it = xlines.find(t.Thread());
+      if (it == xlines.end()) {
+        it = xlines.insert(std::make_pair(t.Thread(), xplane->add_lines()))
+             .first;
+        auto* xline = it->second;
+        xline->set_id(t.Thread());
+        xline->set_display_id(t.Thread());
+        xline->set_name(worker.GetThreadName(
+            worker.DecompressThread(t.Thread())));
+        xline->set_display_name(xline->name());
+        //xline->set_timestamp_ns();
+        //xline->set_duration_ps();
+      }
+
+      auto* event = it->second->add_events();
+      event->set_metadata_id(z.first);
+      event->set_offset_ps(t.Zone()->GpuStart() * 1000);
+      event->set_duration_ps(
+          (t.Zone()->GpuEnd() - t.Zone()->GpuStart()) * 1000);
+    }
+  }
+
+  return xspace;
+}
+
+}  // namespace
+
+int main(int argc, char** argv) {
+  // IREE tries to parse all flags. Reduce argc for parsing to exclude
+  // subprogram's arguments.
+  int argc_for_parse = 1;
+  while (argc_for_parse < argc && argv[argc_for_parse][0] == '-') {
+    ++argc_for_parse;
+  }
+  int argc_for_subprogram = argc - argc_for_parse;
+  char** argv_for_subprogram = argv + argc_for_parse;
+
+  iree_flags_parse_checked(IREE_FLAGS_PARSE_MODE_DEFAULT, &argc_for_parse,
+                           &argv);
+
+  std::unique_ptr<tracy::Worker> worker;
+  // Build a worker from a tracy file,
+  if (argc_for_subprogram == 0) {
```

Comment:
I think having a separate tool like `iree-prof-convert` is simpler. Also, then, we can use `iree-prof` for the current program name, which does:

1. help launch tracy profiler.
2. save the output as tracy profile data, text, and xplane.

In addition, we can provide `iree-prof-convert` to convert tracy to the other formats.

Reply:
Done